contracts/token/ERC1155/ERC1155.sol

```diff
@@ -145,6 +145,59 @@ contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI {
         _safeBatchTransferFrom(from, to, ids, amounts, data);
     }
 
+    /**
+     * @dev Transfers `amount` tokens of token type `id` from `from` to `to`. Will mint (or burn) if `from` (or `to`) is the zero address.
+     *
+     * Emits a {TransferSingle} event if only one transfer is done, and {TransferBatch} event for multiple transfer operations.
+     *
+     * Requirements:
+     *
+     * - If `to` refers to a smart contract, it must implement {IERC1155Receiver-onERC1155BatchReceived} and return the
+     * acceptance magic value.
+     */
+    function _update(
+        address from,
+        address to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        bytes memory data
+    ) internal virtual {
+        require(ids.length == amounts.length, "ERC1155: ids and amounts length mismatch");
+
+        address operator = _msgSender();
+
+        for (uint256 i = 0; i < ids.length; ++i) {
+            uint256 id = ids[i];
+            uint256 amount = amounts[i];
+
+            uint256 fromBalance = _balances[id][from];
+            if (to == address(0)) {
+                require(fromBalance >= amount, "ERC1155: burn amount exceeds balance");
+            }
+            if (from != address(0)) {
+                require(fromBalance >= amount, "ERC1155: insufficient balance for transfer");
+                unchecked {
+                    _balances[id][from] = fromBalance - amount;
+                }
+            }
```

Comment:
There is a check duplication, that is also an actual error:
- we only need to check the from balance if from an account (not if its 0x0)
- if from is 0, then we mint. In that case fromBalance is 0, but we don't care that is less then the amount.
- Since `fromBalance` is only necessary if from is not 0x0, we should only do the load inside this block


```suggestion
            if (from != address(0)) {
                uint256 fromBalance = _balances[id][from];
                require(fromBalance >= amount, "ERC1155: insufficient balance for transfer");
                unchecked {
                    _balances[id][from] = fromBalance - amount;
                }
            }
```